### PR TITLE
Kozmaz87/validation

### PR DESCRIPTION
**Pull request by kozmaz87**
This will affect:
#85 #72 

Once the pull request for the rest buggy test is merged this will pass

### CODE REVIEW

**Comment by kozmaz87** [comment]
Ok guys. Have fun reviewing this. I am adding some extra tests and obviously some more documentation to it because this is dark magic here. In short the library(PyXB) did not offer out of the box the nice propagation of a parsing context so using python's threadLocal objects I monkeypatched the processing pipeline in the appropriate places to get it do what I wanted it to do for me. @gilbrechbuhler if you don't mind it would be very good for you to take a look and give me an impression how difficult it is to follow what's happening here. There could be an easier solution created if we forked PyXB and added the context object in the framework itself.

**Comment by nigelmegitt** [comment]
Just looked, and for me anyway it does seem hard to follow initially. Yes please to more documentation!

**Comment by gilbrechbuhler** [comment]
I think I understand most of it. However, after one month, I'm starting to get comfortable with pyxb and how we make all that work, so it makes this new part easier to understand. So I think some doc would be great indeed.

**Comment by nigelmegitt** [comment]
Reminder: there are conflicts on this branch that need to be resolved before it can be merged.

**Comment by nigelmegitt** [comment]
@kozmaz87 I think you're still working on this? I don't want to merge if you're going to make some more test or documentation commits.

**Comment by nigelmegitt** [comment]
Merging this as "good enough" for now.
